core/toolbox/toolbox_item.js

```diff
@@ -0,0 +1,224 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview An item in the toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+'use strict';
+
+goog.provide('Blockly.CollapsibleToolboxItem');
+goog.provide('Blockly.SelectableToolboxItem');
+goog.provide('Blockly.ToolboxItem');
+
+goog.requireType('Blockly.IToolbox');
+goog.requireType('Blockly.utils.toolbox');
+goog.requireType('Blockly.WorkspaceSvg');
+
+/**
+ * Class for an item in the toolbox.
+ * @param {!Blockly.utils.toolbox.ToolboxItemJson} toolboxItemDef The JSON defining the
+ *     toolbox item.
+ * @param {!Blockly.IToolbox} toolbox The toolbox that holds the toolbox item.
+ * @constructor
+ */
+Blockly.ToolboxItem = function(toolboxItemDef, toolbox) {
+
+  /**
+   * The id for the category.
+   * @type {string}
+   * @protected
+   */
+  this.id_ = toolboxItemDef['id'] || Blockly.utils.genUid();
```

Comment:
You'll want to use `Blockly.utils.IdGenerator.getNextUniqueId()` here instead of `genUid` as you're using this generated id as a DOM element id attribute below. 

The difference between `IdGenerator` and `genUid` is that IdGenerator returns an id suitable for a DOM id, and matches the spec: https://www.w3.org/TR/html401/types.html#type-name